services/web-app/utils/mdx-image-optimizer-plugin.mjs

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright IBM Corp. 2022, 2022
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import { getPlaiceholder } from 'plaiceholder'
+import { visit } from 'unist-util-visit'
+
+const mdxImageOptimizerPlugin = (dirPath) => async (tree) => {
```

Comment:
It might be good to document the assumptions and the contract of this plugin. Based on the presence of `raw=true` below, it seems like the intent might be to have this only work with github images, so if that's the case I think it's important to document all of that and make the contract of this function crystal clear.

Reply:
Added comments on top of declaration !

Reply:
Can we convert them to jsdoc? :) 
That might entail breaking this up into separate functions. Right now it's a function that returns a function that modifies the AST. That might be hard to jsdoc as-is.

Reply:
Done!!